sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py

```diff
@@ -228,108 +193,245 @@ class ServiceBusError(AzureError):
+                condition = self._condition
+            message = message + "\nError condition: {}".format(str(condition))
+        if self._status_code is not None:
+            message = message + "\nStatus Code: {}".format(str(self._status_code))
+        super(ServiceBusError, self).__init__(message, *kwargs, **kwargs)
 
 
 class ServiceBusConnectionError(ServiceBusError):
     """An error occurred in the connection."""
-
-
-class ServiceBusAuthorizationError(ServiceBusError):
-    """An error occurred when authorizing the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred in the connection."
+        super(ServiceBusConnectionError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
 class ServiceBusAuthenticationError(ServiceBusError):
     """An error occurred when authenticate the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred when authenticate the connection."
+        super(ServiceBusAuthenticationError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
-class NoActiveSession(ServiceBusError):
-    """No active Sessions are available to receive from."""
+class UnauthorizedAccessError(ServiceBusError):
+    """An error occurred when authorizing the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred when authorizing the connection."
+        super(UnauthorizedAccessError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
 class OperationTimeoutError(ServiceBusError):
     """Operation timed out."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "Operation timed out."
+        super(OperationTimeoutError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=False,
+            **kwargs
+        )
 
 
-class ServiceBusMessageError(ServiceBusError):
-    """An error occurred when an operation on a message failed because the message is in an incorrect state."""
-
-
-class MessageContentTooLarge(ServiceBusMessageError, ValueError):
+class MessageSizeExceededError(ServiceBusError, ValueError):
     """Message content is larger than the service bus frame size."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "Message content is larger than the service bus frame size."
+        super(MessageSizeExceededError, self).__init__(
+            message,
+            retryable=False,
+            shutdown_handler=False,
+            **kwargs
+        )
 
 
-class MessageAlreadySettled(ServiceBusMessageError):
+class MessageAlreadySettled(ValueError):
```

Comment:
```suggestion
# ValueError was preferred to ServiceBusMessageError/ServiceBusError as a base
# since this arises only from client validation.
class MessageAlreadySettled(ValueError):
```

Feel free to take it or leave it, but as with the other comment, I'm just big on "writing down why things that are different are different so that those who come across this after understand"